driver-core/src/main/java/com/datastax/driver/core/PagingState.java

```diff
@@ -85,7 +85,8 @@ private PagingState(byte[] complete) {
                 //it is a RegularStatement since Batch statements are not allowed
                 RegularStatement rs = (RegularStatement)statement;
                 md.update(rs.getQueryString().getBytes());
-                values = rs.getValues(protocolVersion);
+                // FIXME use per-cluster configured CodecRegistry
+                values = rs.getValues(protocolVersion, new CodecRegistry());
```

Comment:
Yes very good point.

There are a few places where I still do not see how to access the cluster's codec registry:

* PagingState.hash()
* ArrayBackedResultSet.fromMessage(), only when session is null
* toString() methods and similar
    * BuiltStatement.toString()
    * BuiltStatement.getQueryString()
* TupleType.of(DataType...) - which is deprecated

The first two are a bit worrying, I need to address them - at least before we go GA.